### ISSUE TITLE
RAD dataset file

### ISSUE THREAD

**Issue by xinyanghuang7** [TIMMY-CHAN/MILE#1]
Wonderful work! Could you provide the relevant files and training and testing files of RAD dataset?
For example, train_labeldir = r'/Dataset1/cjw/VQA_RAD/train.jsol'
test_labeldir = r'/Dataset1/cjw/VQA_RAD/test.jsol' and train_rad.py.

**Comment by TIMMY-CHAN** [comment]
感谢您对我们工作的关注，我们对对您导致的困惑表示抱歉。
如果您想要使用VQA-RAD源数据集，您可以直接去其官网进行下载，由于其license我们无法直接提供源数据。
如果您想使用instruction-format的VQA-RAD数据集，我们之前将其整合进了intrustion.json中。我们会在近两天对数据进行更新，单独发布instrucion-format的VQA-RAD和Slake数据集.

best wish

**Comment by xinyanghuang7** [comment]
> 感谢您对我们工作的关注，我们对对您导致的困惑表示抱歉。
> 
> 如果您想要使用VQA-RAD源数据集，您可以直接去其官网进行下载，由于其license我们无法直接提供源数据。
> 
> 如果您想使用instruction-format的VQA-RAD数据集，我们之前将其整合进了intrustion.json中。我们会在近两天对数据进行更新，单独发布instrucion-format的VQA-RAD和Slake数据集.
> 
> 
> 
> best wish

十分感谢你们如此及时的回复！

对你们的工作十分感兴趣，并且十分渴望复现出完整的结果！

很期待你们能够发布所对应的数据集（应该是类似在MISS的代码中提到的train.jsol和test.jsol？），
以及如果需要读取发单独发布的数据集是否需要将dataloader也进行修改？

祝好！

**Comment by TIMMY-CHAN** [comment]
对于非instruction-format的SFT数据，训练集测试集与Slake，VQA-RAD源数据集完全相同。
对于本文构造的instruction-format数据，我们将分别发布slake-instruction和rad-instruction，如果您想进行测试则仍需下载源数据集。我们将会在这段时间发布部分微调后的检查点以方便复现不同微调范式得到的模型的结果。

**Comment by xinyanghuang7** [comment]
十分感谢您及时的回复！通过您的回复，我理解了很多问题，但同时还剩下最后一些问题，在这里为打扰您感到抱歉：

1、对于slake数据集，其源文件中已经包含了train和test的json文件，因此，在训练和测试阶段直接读取即可。但是对于VQA-RAD数据集，我所访问和下载的数据中并没有具体的train和test的json文件（见截图）。因此我会产生一些疑惑，即**VQA-RAD数据集的train和test的json文件是如何划分的**？以及如何获取？

![image](https://github.com/TIMMY-CHAN/MILE/assets/75032525/4bd64258-5664-4818-831e-7e558a2ebdb6)


2、我尝试在完整的intrustion.json中对模型进行训练：

train_datasets = VQA_Instruction_dataset(data_dir=datadir,label_dir=train_labeldir,transform=transform_train)
    train_loader = DataLoader(dataset=train_datasets,batch_size=20,shuffle=True, num_workers=26, drop_last=False)

结合您所提供的代码，我成功运行了脚本，但是如果我想测试模型的效果，我应该**如何指定eval_vqa.py中的datadir = r'.../.../.../Slake/Slake1.0/imgs'和test_labeldir = r'.../.../.../Slake/Slake1.0/slake_test.jsol'路径**？即，我在完整的intrustion.json中进行训练后在哪里测试能够得到完整的论文中不同数据集的结果？

3、在我执行模型训练的过程中，代码需要我指定vqa.yaml中的pretrained: False，但我并不知道**这个预训练模型从哪里获取**？因此我设置成了False，请问这会对训练有些影响吗？所说的对于MISS模型进行微调，那么**baseline model MISS的预训练模型如何获取**？代码所指定的预训练权重指的是这个模型的权重吗？以下是我训练的截图：
![image](https://github.com/TIMMY-CHAN/MILE/assets/75032525/2774015a-17c8-4ff8-b3e6-39f8f889c59b)


十分抱歉对代码有很多疑惑，请您理解！

**Comment by TIMMY-CHAN** [comment]
关于RAD训练与测试集的划分，可以阅读原论文以及官方json中的标注（官方的json文件中测试数据的value中带有test字样）。关于eval_vqa，您需要在下载slake数据集之后将图像路径和测试数据路径更改为您自己的路径。

关于预训练模型，很抱歉我们在短期内暂时不会发布MISS_pretrained的版本（预训练模型消耗了大量资源以及私有数据，一些新的工作正在基于该checkpoint开展），而是在近段时间直接发布'微调完成'的MISS模型和部分MILE模型，您可以直接利用我们发布的checkpoint进行推理·，无需任何训练以快速验证复现不同PEFT范式对模型的影响。或是按照MISS论文中的方法使用相关数据进行重新预训练。

关于已经发布的instruction.json，您可以通过我们的原论文阅读更多instruction-format data的用途。

关于训练的影响，没有经过预训练的模型直接微调将会造成性能的损害。

**Comment by TIMMY-CHAN** [comment]
如果您确实想在短期内基于我们的工作进行拓展，您可以考量在将来使用我们已经微调完成的模型进行增量微调或继续训练。

best wish

**Comment by xinyanghuang7** [comment]
了解了，期待您后续的工作，十分感谢！

**Comment by xinyanghuang7** [comment]
单独发布instrucion-format的VQA-RAD和Slake数据集会分为训练集json和测试集json吗？十分期待，请问大概什么时候发布？

**Comment by TIMMY-CHAN** [comment]
指令格式数据用于对模型进行指令微调，因此不存在指令格式的测试数据，本工作使用的测试数据与以slake 或rad为benchmark的工作是一致的，皆为原数据。
7月底前，我们将会公开我们准备公开的全部资源
感谢您的关注
best wish
